### ISSUE TITLE
[YARA] Error in message processing

### ISSUE THREAD

**Issue by dominictory** [OpenCTI-Platform/connectors#1703]
## Description

`ERROR Error in message processing, reporting error to API | timestamp=2024-01-16T15:06:17.644100Z name=YARA exc_info=Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pycti/connector/opencti_connector_helper.py", line 275, in _data_handler
    message = self.callback(json_data["event"])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/opencti-yara/main.py", line 132, in _process_message
    self._scan_artifact(artifact, yara_indicators)
  File "/opt/opencti-yara/main.py", line 86, in _scan_artifact
    artifact_contents = self._get_artifact_contents(artifact)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/opencti-yara/main.py", line 50, in _get_artifact_contents
    file_id = artifact["importFiles"][0]["id"]
              ~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range`

## Environment

Ubuntu 22.04
OpenCTI 5.12.15
Frontend

## Reproducible Steps

Be on 5.12.15
Add Yara connector docker-compose to standalone docker-compose with other connectors and main platform components
Start stack

## Expected Output

YARA connector enriches artifacts

## Actual Output

No enriching

## Additional information

```INFO Message ack | timestamp=2024-01-16T15:06:16.415491Z name=YARA attributes={"tag":1130}

INFO Reporting work update_received | timestamp=2024-01-16T15:06:16.415932Z name=api attributes={"work_id":"work_a2695e04-6d3d-4496-8ba1-7ed9e13779ea_2024-01-16T15:06:15.958Z"}

INFO Enriching artifact--dec32a45-4591-5dc5-8aef-1680e460d75e | timestamp=2024-01-16T15:06:16.623797Z name=YARA

INFO Reading StixCyberObservable | timestamp=2024-01-16T15:06:16.623978Z name=api attributes={"id":"artifact--dec32a45-4591-5dc5-8aef-1680e460d75e"}

INFO Listing Indicators with filters | timestamp=2024-01-16T15:06:16.686605Z name=api attributes={"filters":"{\"mode\": \"and\", \"filters\": [{\"key\": \"pattern_type\", \"values\": [\"yara\"]}], \"filterGroups\": []}"}

ERROR Error in message processing, reporting error to API | timestamp=2024-01-16T15:06:17.644100Z name=YARA exc_info=Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pycti/connector/opencti_connector_helper.py", line 275, in _data_handler
    message = self.callback(json_data["event"])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/opencti-yara/main.py", line 132, in _process_message
    self._scan_artifact(artifact, yara_indicators)
  File "/opt/opencti-yara/main.py", line 86, in _scan_artifact
    artifact_contents = self._get_artifact_contents(artifact)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/opencti-yara/main.py", line 50, in _get_artifact_contents
    file_id = artifact["importFiles"][0]["id"]
              ~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

INFO Reporting work update_processed | timestamp=2024-01-16T15:06:17.644417Z name=api attributes={"work_id":"work_a2695e04-6d3d-4496-8ba1-7ed9e13779ea_2024-01-16T15:06:15.958Z"}```

**Comment by nino-filigran** [comment]
@mattreduce I think you were the one who created this connecotr. Would you mind having a look at it?

**Comment by dominictory** [comment]
> @mattreduce I think you were the one who created this connecotr. Would you mind having a look at it?

Hi, is there any update on this at all? Keen to get Yara connector working.

**Comment by mattreduce** [comment]
Hi @nino-filigran @dominictory I'll take a look now 👍

**Comment by mattreduce** [comment]
Thank you @nino-filigran and @dominictory! https://github.com/OpenCTI-Platform/connectors/pull/1804 should resolve this issue once accepted and merged. It appears all connectors that work with file contents of Artifacts need to update their code to continue working properly, which was resolved for most connectors with https://github.com/OpenCTI-Platform/connectors/commit/1b8f610249ad04c0d87981052bc5269f58b9ec28. ✅

**Comment by dominictory** [comment]
> Thank you @nino-filigran and @dominictory! #1804 should resolve this issue once accepted and merged. It appears all connectors that work with file contents of Artifacts need to update their code to continue working properly, which was resolved for most connectors with [1b8f610](https://github.com/OpenCTI-Platform/connectors/commit/1b8f610249ad04c0d87981052bc5269f58b9ec28). ✅

Hey, just wanted to let you know, on 6.3.1 I am seeing the below:

```
{"timestamp": "2024-09-24T09:14:12.027182Z", "level": "ERROR", "name": "YARA", "message": "Error in message processing, reporting error to API", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.11/site-packages/pycti/connector/opencti_connector_helper.py\", line 352, in _data_handler\n    message = self.callback(event_data)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/opencti-yara/main.py\", line 107, in _process_message\n    self._scan_artifact(artifact, yara_indicators)\n  File \"/opt/opencti-yara/main.py\", line 63, in _scan_artifact\n    artifact_contents = self._get_artifact_contents(artifact)\n                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/opencti-yara/main.py\", line 27, in _get_artifact_contents\n    file_id = artifact[\"importFiles\"][0][\"id\"]\n              ~~~~~~~~~~~~~~~~~~~~~~~^^^\nIndexError: list index out of range"}
```
